packages/now-go/index.ts

```diff
@@ -436,4 +460,111 @@ Learn more: https://vercel.com/docs/runtimes#official-runtimes/go
   };
 }
 
-export { shouldServe };
+function isPortInfo(v: any): v is PortInfo {
+  return v && typeof v.port === 'number';
+}
+
+function isReadable(v: any): v is Readable {
+  return v && v.readable === true;
+}
+
+async function copyEntrypoint(entrypoint: string, dest: string): Promise<void> {
+  const data = await readFile(entrypoint, 'utf8');
+
+  // Modify package to `package main`
+  const patched = data.replace(/\bpackage\W+\S+\b/, 'package main');
+
+  await writeFile(join(dest, 'entrypoint.go'), patched);
+}
+
+async function copyDevServer(
+  functionName: string,
+  dest: string
+): Promise<void> {
+  const data = await readFile(join(__dirname, 'dev-server.go'), 'utf8');
+
+  // Populate the handler function name
+  const patched = data.replace('__HANDLER_FUNC_NAME', functionName);
+
+  await writeFile(join(dest, 'vercel-dev-server-main.go'), patched);
+}
+
+export async function startDevServer(
+  opts: StartDevServerOptions
+): Promise<StartDevServerResult> {
+  const { entrypoint, workPath, meta = {} } = opts;
+  const { devCacheDir = join(workPath, '.vercel', 'cache') } = meta;
+  const entrypointDir = dirname(entrypoint);
+
+  const tmp = join(
+    devCacheDir,
+    'go',
+    Math.random()
+      .toString(32)
+      .substring(2)
+  );
+  const tmpPackage = join(tmp, entrypointDir);
+  await mkdirp(tmpPackage);
+
+  let goModAbsPathDir = '';
+  if (await pathExists(join(workPath, 'go.mod'))) {
+    goModAbsPathDir = workPath;
+  }
+  const analyzedRaw = await getAnalyzedEntrypoint(entrypoint, goModAbsPathDir);
+  if (!analyzedRaw) {
+    throw new Error(
```

Comment:
Can this throw `NowBuildError` instead?

Reply:
It was copy/pasted from the `build()` function. I think we should do it in a follow up PR since it'll require a `docs` change as well.